source/codegen/validate_examples.py

```diff
@@ -0,0 +1,70 @@
+from argparse import ArgumentParser
+from contextlib import contextmanager
+from os import system, getcwd, chdir
+from pathlib import Path
+from shutil import rmtree
+
+from stage_client_files import stage_client_files
+
+@contextmanager
+def _create_stage_dir(staging_dir):
+  initial_dir = getcwd()
+  try:
+    rmtree(staging_dir, ignore_errors=True)
+    staging_dir.mkdir(parents=True)
+    chdir(staging_dir)
+    yield
+  finally:
+    chdir(initial_dir)
+
+def validate_examples(driver_glob_expression: str, ip_address: str) -> None:
+  staging_dir = Path(__file__).parent.parent.parent / "build" / "validate_examples"
+
+  print(f"Validating examples using staging_dir: {staging_dir}")
+
+  with _create_stage_dir(staging_dir):
+    system("poetry new .")
+    system("poetry add grpcio")
+    system("poetry add --dev grpcio-tools black mypy mypy-protobuf types-protobuf grpc-stubs")
+    system("poetry install")
+
+    stage_client_files(staging_dir, True)
+    examples_dir = staging_dir / "examples"
+    proto_dir = staging_dir / "proto"
+    driver_example_dirs = list(examples_dir.glob(driver_glob_expression))
+    driver_proto_names = [f"{dir.name}.proto" for dir in driver_example_dirs]
+    proto_files_str = str.join(" ", driver_proto_names + ["session.proto", "nidevice.proto"])
+
+    system(
+      rf"poetry run python -m grpc_tools.protoc -I{proto_dir} --python_out=. --grpc_python_out=. --mypy_out=. --mypy_grpc_out=. {proto_files_str}"
+    )
+    for dir in driver_example_dirs:
+      print()
+      print(f"-- Validating: {dir.name} --")
+      
+      print(f" -> Running black line-length 100")
+      system(f"poetry run black --check --line-length 100 {dir}")
```

Comment:
I'm guessing we need to tweak this to support either (a) failing workflow runs or (b) auto-updating examples in place (or general local iterative testing while developing examples). But this is a decent start for a semi-automated visual inspection.

Reply:
`system()` returns the process exitcode (on all reasonable systems). So you can save that off and use it to determine whether you might also want to sys.exit(1) at the end of the script.

Reply:
I'll plan on using this solution in a separate change that hopefully can get mypy checking added to the PR workflow.